src/main/java/org/jenkinsci/modules/windows_slave_installer/SlaveExeUpdater.java

```diff
@@ -49,10 +50,18 @@ public void onOnline(Computer c, final TaskListener listener) throws IOException
 
         final SlaveComputer sc = (SlaveComputer) c;
 
+        try{
+            if(sc.isUnix()) return;
+        }catch (NullPointerException ex){
```

Comment:
Why does it check it in such way. Which method does throw the NPE?

Reply:
Ah I see. `isUnix` returns Boolean.

A correct code would be...

```java
final Boolean isUnix = sc.isUnix()
if (isUnix == null || isUnix) { // Do not try installing on disconnected or Unix machines
  return;
}
```

Disconnected may really happen here in the case of race condition